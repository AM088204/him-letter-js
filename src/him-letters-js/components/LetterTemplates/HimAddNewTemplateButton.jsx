import React from 'react';
import { DisclosureManagerContext } from 'terra-disclosure-manager';
import ModalManager from 'terra-modal-manager';
import IconAdd from 'terra-icon/lib/icon/IconAdd';
import Button from 'terra-button/lib/Button';
import { injectIntl } from 'react-intl';
import HimAddOrEditWindow from './HimAddOrEditWindow';
import '../../stylesheets/HimLetterTemplates.css';

const HimAddNewTemplateButton = (props) => {
  const { disclosureType, size ,intl } = props;
  const disclosureManager = React.useContext(DisclosureManagerContext);
  
  return (
    <div>
      <ModalManager >
        <Button
          text={intl.formatMessage({ id: 'him-letters-engine.himAddNewTemplate.button' })}
          icon={<IconAdd />}
          variant="emphasis"
          onClick={() => {
            disclosureManager.disclose({
              preferredType: disclosureType,
              size,
              content: {
                key: "add new template",
                component: <HimAddOrEditWindow name={intl.formatMessage({ id: 'him-letters-engine.himAddNewTemplate.button' })} disclosureType={disclosureType} />,
             },
            });
          }}
        >
      </Button>      
    </ModalManager>
   </div>);
}
export default injectIntl(HimAddNewTemplateButton);