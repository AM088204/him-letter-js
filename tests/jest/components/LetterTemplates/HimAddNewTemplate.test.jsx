import React from 'react';
import { shallowWithIntl, mountWithIntl } from 'terra-enzyme-intl';
import HimAddNewTemplateButton from '../../../../src/him-letters-js/components/LetterTemplates/HimAddNewTemplateButton';
import Button from 'terra-button/lib/Button';
import { DisclosureManagerContext } from 'terra-disclosure-manager';
import MockContext from './MockContext';
describe('HimAddNewTemplate', () => {
    const disclosureSpy = jest.fn();
    beforeAll(() => {
        MockContext({ disclose: disclosureSpy });
    });
    it('Should Render HimAddNewTemplate Button', () => {
        const himAddNewTemplate = shallowWithIntl(<HimAddNewTemplateButton />);
        expect(himAddNewTemplate.first().shallow()).toMatchSnapshot();
    });
    it('Should verify the label', () => {
        const himAddNewTemplate = mountWithIntl(
            <HimAddNewTemplateButton />
        );
        expect(himAddNewTemplate.find(Button).props().text).toBe('him-letters-engine.himAddNewTemplate.button');
    });

    it('Should verify the variant', () => {
        const himAddNewTemplate = mountWithIntl(
            <HimAddNewTemplateButton />
        );
        expect(himAddNewTemplate.find(Button).props().variant).toBe('emphasis');
    });

    it('Should verify the add icon', () => {
        const himAddNewTemplate = mountWithIntl(
            <HimAddNewTemplateButton />
        );
        expect(himAddNewTemplate.find(Button).props().icon.type.displayName).toBe('IconAdd');
    });

    it('Should verify the onclick', () => {
        const mockClick = jest.fn();
        const himAddNewTemplate = mountWithIntl(
            <HimAddNewTemplateButton disclosureType="modal" size="large" />
        );
        himAddNewTemplate
            .find(Button)
            .props()
            .onClick(mockClick);

        console.log("dispatchnnnnnn", disclosureSpy.mock.calls[0][0]);
    });
});