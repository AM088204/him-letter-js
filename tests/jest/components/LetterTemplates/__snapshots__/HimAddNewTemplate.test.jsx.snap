// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HimAddNewTemplate Should Render HimAddNewTemplate Button 1`] = `
<div>
  <ModalManager>
    <Button
      icon={
        <IconAdd
          data-name="Layer 1"
          viewBox="0 0 48 48"
          xmlns="http://www.w3.org/2000/svg"
        />
      }
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isIconOnly={false}
      isReversed={false}
      onClick={[Function]}
      text="him-letters-engine.himAddNewTemplate.button"
      type="button"
      variant="emphasis"
    />
  </ModalManager>
</div>
`;
